charts/redpanda/templates/_helpers.tpl

```diff
@@ -538,6 +538,10 @@ than 1 core.
 {{- toJson (dict "bool" (or (not (eq .Values.image.repository "docker.redpanda.com/redpandadata/redpanda")) (include "redpanda.semver" . | semverCompare ">=22.2.10-0,<22.3"))) -}}
 {{- end -}}
 
+{{- define "redpanda-22-2-x-without-sasl" -}}
+{{- toJson (dict "bool" (or (not (or (include "sasl-enabled" . | fromJson).bool .Values.listeners.kafka.authenticationMethod)) (include "redpanda-atleast-22-3-0" . | fromJson).bool)) -}}
```

Comment:
If this is accurate, I'd suggest this. The nested logic is way to hard to parse in my head.

```suggestion
{{- $result :=  (include "redpanda-atleast-22-3-0" . | fromJson).bool -}}
{{- if or (include "sasl-enabled" . | fromJson).bool .Values.listeners.kafka.authenticationMethod)) -}}
{{-   $result := false -}}
{{- end -}}
{{- toJson (dict "bool" $result -}}
```

Reply:
I'm not that proficient in templating langue. If I could code this in go :)

Comment included into next push

Reply:
It IS go! :grin: